src/main/java/io/jenkins/plugins/gcr/models/PluginEnvironment.java

```diff
@@ -3,38 +3,35 @@
 import hudson.EnvVars;
 import hudson.model.Run;
 
-public class PluginEnvironment {
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
-    private String pullRequestId;
+public class PluginEnvironment {
 
     private String pullRequestRepository;
 
-    private String gitUrl;
-
     private String gitHash;
 
     private String buildUrl;
 
     // Constructor
 
     public PluginEnvironment(EnvVars env) throws IllegalArgumentException {
-        pullRequestId = get("ghprbPullId", env);
-        pullRequestRepository = get("ghprbGhRepository", env);
-        gitUrl = get("ghprbAuthorRepoGitUrl", env);
-        gitHash = get("ghprbActualCommit", env);
+    	String changeUrl = get("CHANGE_URL", env);
+
+		Pattern pattern = Pattern.compile("https://github.com/(.*?)/pull/.*");
```

Comment:
This won't work with Github Enterprise servers, which I require to be supported.

Reply:
mmh, true! I can try to fix that if you want

Reply:
I use GitHub Enterprise as well.

Reply:
so something like
```https://[^/]*/([^/]*?)/pull/.*```

Reply:
(or `https://[^/]*/(.*?)/pull/.*` but I prefer the other one)

Reply:
I pushed a commit.
But can you validate also that this regex is working for you (for example here: https://regex101.com/)